push-down-test/run-tests.sh

```diff
@@ -0,0 +1,229 @@
+readonly no_push_down_tidb_log_file="/tmp/copr_test/tidb_no_push_down.log"
+readonly push_down_no_batch_tidb_log_file="/tmp/copr_test/tidb_push_downno_batch.log"
+readonly push_down_with_batch_tidb_log_file="/tmp/copr_test/tidb_push_downwith_batch.log"
+readonly tikv_no_batch_log_file="/tmp/copr_test/tikv_no_batch.log"
+readonly tikv_with_batch_log_file="/tmp/copr_test/tikv_with_batch.log"
+readonly pd_no_batch_log_file="/tmp/copr_test/pd_no_batch.log"
+readonly pd_with_batch_log_file="/tmp/copr_test/pd_with_batch.log"
+
+readonly no_push_down_config_dir="./config/no_push_down"
+readonly push_down_no_batch_config_dir="./config/push_down_no_batch"
+readonly push_down_with_batch_config_dir="./config/push_down_with_batch"
+
+readonly push_down_test_bin="push_down_test_bin"
+
+export GO111MODULE=on
+export GOPROXY=https://goproxy.io
+
+set -u
+
+function run_pds() {
+  "$pd_bin" -config "$push_down_no_batch_config_dir"/pd.toml -log-file "$pd_no_batch_log_file" -L ${log_level} &
+  pd_no_batch_pid=$!
+
+  "$pd_bin" -config "$push_down_with_batch_config_dir"/pd.toml -log-file "$pd_with_batch_log_file" -L ${log_level} &
+  pd_with_batch_pid=$!
+}
+
+function run_tikvs() {
+  "$tikv_bin" -C "$push_down_no_batch_config_dir"/tikv.toml --log-file "$tikv_no_batch_log_file" -L ${log_level} &
+  tikv_no_batch_pid=$!
+
+  "$tikv_bin" -C "$push_down_with_batch_config_dir"/tikv.toml --log-file "$tikv_with_batch_log_file" -L ${log_level} &
+  tikv_with_batch_pid=$!
+}
+
+function run_tidbs() {
+  export GO_FAILPOINTS="github.com/pingcap/tidb/expression/PushDownTestSwitcher=return(\"\")"
+  "$tidb_bin" -log-file "$no_push_down_tidb_log_file" -config "$no_push_down_config_dir"/tidb.toml -L ${log_level} &
+  tidb_no_push_down_pid=$!
+
+  export GO_FAILPOINTS="github.com/pingcap/tidb/expression/PushDownTestSwitcher=return(\"all\")"
+  "$tidb_bin" -log-file "$push_down_no_batch_tidb_log_file" -config "$push_down_no_batch_config_dir"/tidb.toml -L ${log_level} &
+  tidb_push_down_no_batch_pid=$!
+
+  export GO_FAILPOINTS="github.com/pingcap/tidb/expression/PushDownTestSwitcher=return(\"all\")"
+  "$tidb_bin" -log-file "$push_down_with_batch_tidb_log_file" -config "$push_down_with_batch_config_dir"/tidb.toml -L ${log_level} &
+  tidb_push_down_with_batch_pid=$!
+}
+
+# make sure that after run this function,
+# the current working dir is not change
+function build_tidb() {
+  if [ -z $tidb_src_dir ]; then
+    echo "TiDB source code path not specified (tidb_src_dir environment variable not set)"
+    readonly tidb_src_url="https://github.com/pingcap/tidb/archive/master.zip"
+    echo "Retrieve TiDB source code from ${tidb_src_url}"
+    mkdir -p /tmp/copr_test/
+    rm -rf /tmp/copr_test/tidb_master /tmp/copr_test/tidb_master.zip
+    wget ${tidb_src_url} -O /tmp/copr_test/tidb_master.zip
+    unzip /tmp/copr_test/tidb_master.zip -d /tmp/copr_test/tidb_master
+    tidb_src_dir=/tmp/copr_test/tidb_master/tidb-master
+  fi
+
```

Comment:
I think we should not specify the default `tidb_src_dir`, because if there is no this env var, it is CI script's bug. Then if we has default value for it, will **hide the problem**, we should fail fast here.

Reply:
We should provide the option to specify the tidb_src_dir, tikv_bin, pd_bin. To make the debug in localhost easy.